plugins/node/opentelemetry-instrumentation-pg/src/instrumentation.ts

```diff
@@ -173,101 +167,107 @@ export class PgInstrumentation extends InstrumentationBase {
         `Patching ${PgInstrumentation.COMPONENT}.Client.prototype.query`
       );
       return function query(this: PgClientExtended, ...args: unknown[]) {
-        let span: Span;
-
-        // Handle different client.query(...) signatures
-        if (typeof args[0] === 'string') {
-          const query = args[0];
-
-          if (args.length > 1 && args[1] instanceof Array) {
-            const params = args[1];
-            span = utils.handleParameterizedQuery.call(
-              this,
-              plugin.tracer,
-              plugin.getConfig(),
-              query,
-              params
-            );
-          } else {
-            span = utils.handleTextQuery.call(
-              this,
-              plugin.tracer,
-              plugin.getConfig(),
-              query
-            );
-          }
+        // client.query(text, cb?), client.query(text, values, cb?), and
+        // client.query(configObj, cb?) are all valid signatures. We construct
+        // a queryConfig obj from all (valid) signatures to build the span in a
+        // unified way. We verify that we at least have query text, and code
+        // defensively when dealing with `queryConfig` after that (to handle all
+        // the other invalid cases, like a non-array for values being provided).
+        // The type casts here reflect only what we've actually validated.
+        const arg0 = args[0];
+        const firstArgIsString = typeof arg0 === 'string';
+        const firstArgIsQueryObjectWithText = isObjectWithTextString(arg0);
+
+        // TODO: remove the `as ...` casts below when the TS version is upgraded.
+        // Newer TS versions will use the result of firstArgIsQueryObjectWithText
+        // to properly narrow arg0, but TS 4.3.5 does not.
+        const queryConfig = firstArgIsString
+          ? { text: arg0 as string, values: args[1] }
```

Comment:
Great catch! 

Technically, the code as-is wouldn't try to add the callback as an attribute (because we check that we're dealing with an array before setting the attribute, to handle cases like `client.query({ values: notAnArray })`), but it would pass it to the hook, and it's better not to do that. I'll push a fix!